test/test_linalg.py

```diff
@@ -183,6 +183,266 @@ def test_det(self, device, dtype):
+            with self.assertRaisesRegex(ValueError, "be \'L\' or \'U\'"):
+                np.linalg.eigh(t.cpu().numpy(), UPLO=uplo)
+
+        # if non-empty out tensor with wrong shape is passed a warning is given
+        a = random_hermitian_matrix(3, dtype=dtype, device=device)
+        real_dtype = a.real.dtype if dtype.is_complex else dtype
+        out_w = torch.empty(7, 7, dtype=real_dtype, device=device)
+        out_v = torch.empty(7, 7, dtype=dtype, device=device)
+        with warnings.catch_warnings(record=True) as w:
+            # Trigger warning
+            torch.linalg.eigh(a, out=(out_w, out_v))
+            # Check warning occurs
+            self.assertEqual(len(w), 2)
+            self.assertTrue("An output with one or more elements was resized" in str(w[-2].message))
+            self.assertTrue("An output with one or more elements was resized" in str(w[-1].message))
+
+        # dtypes should match
+        out_w = torch.empty_like(a).to(torch.int)
+        out_v = torch.empty_like(a)
+        with self.assertRaisesRegex(RuntimeError, "dtype Int does not match self dtype"):
+            torch.linalg.eigh(a, out=(out_w, out_v))
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @precisionOverride({torch.float32: 1e-4, torch.complex64: 1e-4})
+    def test_eigh_non_contiguous(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_matrix
+
+        def run_test(matrix, uplo):
+            self.assertFalse(matrix.is_contiguous())
+            expected_w, expected_v = np.linalg.eigh(matrix.cpu().numpy(), UPLO=uplo)
+            actual_w, actual_v = torch.linalg.eigh(matrix, UPLO=uplo)
+            self.assertEqual(actual_w, expected_w)
+            # sign of eigenvectors is not unique and therefore absolute values are compared
+            self.assertEqual(abs(actual_v), abs(expected_v))
+
+        def run_test_permuted(shape, batch, uplo):
+            # check for permuted / transposed inputs
+            matrix = random_hermitian_matrix(shape, *batch, dtype=dtype, device=device)
+            matrix = matrix.transpose(-2, -1)
+            run_test(matrix, uplo)
+
+        def run_test_skipped_elements(shape, batch, uplo):
+            # check for inputs with skipped elements
+            matrix = random_hermitian_matrix(shape, *batch, dtype=dtype, device=device)
+            matrix = matrix[::2]
+            run_test(matrix, uplo)
+
+        shapes = (3, 5)
+        batches = ((4, ), (4, 2))
+        uplos = ["U", "L"]
+        for shape, batch, uplo in itertools.product(shapes, batches, uplos):
+            run_test_permuted(shape, batch, uplo)
+            run_test_skipped_elements(shape, batch, uplo)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float64, torch.complex128)
+    @dtypesIfCUDA(torch.float64)
+    def test_eigh_autograd(self, device, dtype):
```

Comment:
Testing of `eigh` differentiation requires a modified loss function, `torch.linalg.eigh` is not tested directly. I think it can't be implemented with `method_tests`.

Reply:
That's OK. We'll soon switch to using OpInfo-based tests and can revisit this.